src/rdkafka.h

```diff
@@ -9850,6 +9906,132 @@ RD_EXPORT void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+ *
+ * @param election_type The election type that needs to be performed,
+ *        preferred or unclean.
+ * @param partitions The topic partitions for which the leader election
+ *        needs to be performed.
+ *
+ * @returns a new allocated elect leaders object or returns NULL in case
+ *          of invalid election_type.
+ *          Use rd_kafka_ElectLeaders_destroy() to free object when done.
+ */
+RD_EXPORT rd_kafka_ElectLeaders_t *
+rd_kafka_ElectLeaders_new(rd_kafka_ElectionType_t election_type,
+                          rd_kafka_topic_partition_list_t *partitions);
+
+/**
+ * @brief Destroy and free a rd_kafka_ElectLeaders_t object previously created
+ *        with rd_kafka_ElectLeaders_new()
+ *
+ * @param elect_leaders The rd_kafka_ElectLeaders_t object to be destroyed.
+ */
+RD_EXPORT void
+rd_kafka_ElectLeaders_destroy(rd_kafka_ElectLeaders_t *elect_leaders);
+
+/**
+ * @brief Elect Leaders for the provided Topic Partitions
+ *        according to the specified election type.
+ *
+ * @param rk Client instance.
+ * @param elect_leaders The elect leaders request containing
+ *        election type and partitions information.
+ * @param options Optional admin options, or NULL for defaults.
+ * @param rkqu Queue to emit result on.
+ *
+ * Supported admin options:
+ *  - rd_kafka_AdminOptions_set_operation_timeout() - default 60 seconds.
+ *    Controls how long the brokers will wait for records to be deleted.
+ *  - rd_kafka_AdminOptions_set_request_timeout() - default socket.timeout.ms.
+ *    Controls how long \c rdkafka will wait for the request to complete.
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_ELECTLEADERS_RESULT
+ */
+RD_EXPORT void rd_kafka_ElectLeaders(rd_kafka_t *rk,
+                                     rd_kafka_ElectLeaders_t *elect_leaders,
+                                     const rd_kafka_AdminOptions_t *options,
+                                     rd_kafka_queue_t *rkqu);
+
+/**
+ * @brief  A struct representing result of elect leaders admin operation.
+ */
+typedef struct rd_kafka_ElectLeadersResult_s rd_kafka_ElectLeadersResult_t;
+
+/**
+ * @brief Get the elect leaders result from the elect leaders result event.
+ *
+ * @param result The elect leaders result event.
+ *
+ * @returns the elect leaders result from the elect leaders result event.
+ */
+RD_EXPORT const rd_kafka_ElectLeadersResult_t *
+rd_kafka_ElectLeaders_result(const rd_kafka_ElectLeaders_result_t *result);
+
+/**
+ * @brief Get error code from the elect leaders result.
+ *
+ * @param result The elect leaders result.
+ *
+ * @returns the error code from the elect leaders result.
+ *
+ * @remark This error is related to the overall elect leaders operation.
+ *         Individual partition related errors are present at partition level.
+ */
+RD_EXPORT rd_kafka_resp_err_t
+rd_kafka_ElectLeadersResult_error(const rd_kafka_ElectLeadersResult_t *result);
```

Comment:
This isn't needed as the top level error needs to be in the event.
```suggestion
RD_EXPORT rd_kafka_resp_err_t
rd_kafka_ElectLeadersResult_error(const rd_kafka_ElectLeadersResult_t *result);
```